### ISSUE TITLE
Rounded borders for RectDecoration

### ISSUE THREAD

**Issue by Crystal4276** [elParaguayo/qtile-extras#193]
I'm trying to modify a bit the look of Rectdecoration to obtain some subtle borders by stacking two decorations in a row.
```python
decor_nogroup = {
    "decorations": [
        RectDecoration(
            colour=colors[9],
            #use_widget_background=True,
            line_width= 0,
            radius=[15, 15, 15, 15],
            filled=True,
            padding_y=10,
            padding_x=0,
            group=False,
        ), 
        RectDecoration(
            colour=colors[0],
            #use_widget_background=True,
            line_width= 1,
            radius=15,
            filled=False,
            padding_y=10,
            padding_x=0,
            group=False,
        ), 
    ],
}
```
Thus, the second decoration is here to play the role of borders with rounded corners.
I'm having issues with `padding_x=0` for the second decoration.
As you can see the "border" is missing in the middle.
![Screenshot from 2022-11-19 09-28-59](https://user-images.githubusercontent.com/27446196/202837889-5d1dfb36-a836-4a89-9792-101a16509115.png)

**Comment by Crystal4276** [comment]
I found another issue when the second Rectdecoration is used with `group=True`:
```python
decor_mem = {
    "decorations": [
        RectDecoration(
            #colour="#a6e3a1",
            use_widget_background=True,
            line_width= 0,
            radius=[15, 15, 15, 15],
            filled=True,
            margin_y=20,
            padding_y=10,
            padding_x=0,
            group=True,
            clip=False,
        ),
        RectDecoration(
            colour=colors[6],
            #use_widget_background=True,
            line_width= 2,
            radius=15,
            filled=False,
            padding_y=10,
            padding_x=0,
            group=True,
        ), 
    ],
}
```
There are some line separations between the three widgets : 
![image](https://user-images.githubusercontent.com/27446196/202783939-e3ab9a86-4554-4a83-9294-e85e9c3cd757.png)

**Comment by elParaguayo** [comment]
It's really hard to follow what you're saying. If these are two different issues then please split it into 2 separate posts.

Also, I'm going to need more of the config. You're showing multiple widgets but it's not clear if they all use the same decoration or not.

**Comment by Crystal4276** [comment]
Code of the widget
```python
                widget.GroupBox(
                       font="monospace",
                       fontsize = 35,
                       spacing = 10,
                       margin_y = 3,
                       margin_x = 15,
                       padding = 5,
                       #background="#45475a",
                       disable_drag = True,
                       active = colors[3],
                       inactive = colors[2],
                       highlight_method='border',
                       this_current_screen_border = colors[3],
                       other_current_screen_border=colors[0],
                       this_screen_border=colors[3],
                       urgent_border=colors[3],
                       urgent_text=colors[3],
                       borderwidth = 3,
                       visible_groups=['1', '2', '3'],
                       **decor_nogroup
                ),
```

**Comment by Crystal4276** [comment]
I'm wondering if adding round corners to BorderDecoration would be an alternative ?
